go.mod

```diff
@@ -1,9 +1,11 @@
 module github.com/hashicorp/terraform-registry-address
 
-go 1.14
+go 1.17
```

Comment:
This [directive](https://go.dev/doc/modules/gomod-ref#go) basically says "the module requires Go MAJOR.MINOR" and it would just affect error reporting when compilation downstream fails.

AFAICT our module is still 1.14 compatible - or do you plan to introduce something that would make it incompatible? I'm assuming the `go.mod` format itself remains backwards compatible for _reading_, otherwise the `go` directive would seem pointless?

Reply:
Since we're currently only testing the pinned Go version in `.go-version`, I assumed it was easier to increase this version than to introduce a test matrix for older versions.

If we stay at `1.14` what's a good way to ensure that every change still works with this version?

Reply:
Hmm, I see your point... yeah - this is easier than introducing a test matrix. Also we tend to support last two major Go versions in most products/libraries, so bumping to 1.17 shouldn't hurt.

Reply:
Keeping support for older Go versions is also a very valid point, so I updated the PR with a test matrix which was easier than expected :)